xmppserver/src/main/java/org/jivesoftware/openfire/OfflineMessageStore.java

```diff
@@ -79,11 +85,44 @@
         "DELETE FROM ofOffline WHERE username=?";
     private static final String DELETE_OFFLINE_MESSAGE =
         "DELETE FROM ofOffline WHERE username=? AND creationDate=?";
+    private static final String DELETE_OFFLINE_MESSAGE_BEFORE =
+            "DELETE FROM ofOffline creationDate < ?";
+    private static final String SELECT_SIZE_OFFLINE_ALL_USERS =
+            "SELECT SUM(messageSize),username FROM ofOffline group by username";
 
     private static final int POOL_SIZE = 10;
-    
+
     private Cache<String, Integer> sizeCache;
 
+    /**
+     * Members for automatic offline message cleaning
+     * */
+
```

Comment:
hmm ok you are semi right ;D
if you change the enable/disable switch in admin console the timer will be restarted with new values... but you are right in case of changing the values in the system values editor

Reply:
i have changed them to be not dynamic